projects/angular-calendar/src/modules/common/keydown.directive.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  Directive,
+  Renderer2,
+  ElementRef,
+  OnInit,
+  OnDestroy,
+  Output,
+  EventEmitter,
+  Inject,
+  Input,
+  NgZone
+} from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { Observable, Subject } from 'rxjs';
+import { requestIdleCallbackObservable } from './request-idle-callback';
+import { switchMapTo, takeUntil } from 'rxjs/operators';
+
+const keydownElements = new Set<HTMLElement>();
+
+const eventName = 'keydown.enter';
+
+@Directive({
+  selector: '[mwlKeydown]'
+})
+export class KeydownDirective implements OnInit, OnDestroy {
+  @Input() keydownListenerDisabled = false;
+
+  @Output('mwlKeydown') keydown = new EventEmitter<KeyboardEvent>(); // tslint:disable-line
+
+  private destroy$ = new Subject();
+
+  constructor(
+    private renderer: Renderer2,
+    private elm: ElementRef<HTMLElement>,
+    @Inject(DOCUMENT) private document,
+    private zone: NgZone
+  ) {}
+
+  ngOnInit(): void {
```

Comment:
I think this can be simplified to:
```
@HostListener('keydown', ['$event'])
onKeyPress(event: KeyboardEvent) {
  if (
            event.keyCode === 13 ||
            event.which === 13 ||
            event.key === 'Enter'
          ) {
            event.preventDefault();
event.stopPropagation();
this.keydown.emit(event);
          }
}
```